tokio/tests/uds_split.rs

```diff
@@ -30,8 +30,8 @@ async fn split() -> std::io::Result<()> {
 }
 
 async fn send_recv_all(
-    read: &mut (dyn AsyncRead + Unpin),
-    write: &mut (dyn AsyncWrite + Unpin),
+    read: &mut (impl AsyncRead + Unpin),
+    write: &mut (impl AsyncWrite + Unpin),
```

Comment:
I assume this change is from when you had troubles with object safety. It should not be necessary.

Reply:
I tried reverting:

```rust
error: the `write_all` method cannot be invoked on a trait object
   --> tokio/tests/uds_split.rs:37:11
    |
37  |     write.write_all(input).await?;
    |           ^^^^^^^^^
    | 
   ::: /home/carllerche/Code/tokio/tokio/src/io/async_write.rs:379:15
    |
379 |         Self: Sized + Unpin,
    |               ----- this has a `Sized` requirement
```

Removing the `Sized` requirement on `fn write_all` prevents `AsyncRead` from becoming a trait object itself.

Reply:
Same problem as here: https://docs.rs/async-trait/0.1.36/async_trait/#dyn-traits

Reply:
This compiles, but is not particularly convenient.
```Rust
async fn send_recv_all(
    mut read: &mut (dyn AsyncRead + Unpin),
    mut write: &mut (dyn AsyncWrite + Unpin),
    input: &[u8],
) -> std::io::Result<Vec<u8>> {
    (&mut write).write_all(input).await?;
    (&mut write).shutdown().await?;

    let mut output = Vec::new();
    (&mut read).read_to_end(&mut output).await?;
    Ok(output)
}
```